### PR TITLE
Add metrics and resource attribute config options from vcentereceiver in docs and tests

### PR DESCRIPTION
**Pull request by marctc**
#### PR Description
After merging #5715 we identified a [couple of things](https://github.com/grafana/agent/pull/5715#pullrequestreview-1734826498) to be done in a follow up PR.
This PR adds the missing configurable fields from vcenter receiver to its documentations site.
It also adds case scenarios for config conversion.

#### PR Checklist

- [ ] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [ ] Config converters updated

### CODE REVIEW

**Review by rfratto — APPROVED** [review]

File: component/otelcol/receiver/vcenter/vcenter_test.go [hunk 1]
```diff
@@ -36,7 +174,53 @@ func TestArguments_UnmarshalRiver(t *testing.T) {
...
+	require.Equal(t, true, otelArgs.ResourceAttributes.VcenterDatastoreName.Enabled)
+	require.Equal(t, true, otelArgs.ResourceAttributes.VcenterHostName.Enabled)
+	require.Equal(t, false, otelArgs.ResourceAttributes.VcenterResourcePoolInventoryPath.Enabled)
+	require.Equal(t, true, otelArgs.ResourceAttributes.VcenterResourcePoolName.Enabled)
+	require.Equal(t, true, otelArgs.ResourceAttributes.VcenterVMName.Enabled)
+	require.Equal(t, true, otelArgs.ResourceAttributes.VcenterVMID.Enabled)
+
+	// Verify MetricsConfig fields
+	require.Equal(t, false, otelArgs.Metrics.VcenterClusterCPUEffective.Enabled)
+	require.Equal(t, true, otelArgs.Metrics.VcenterClusterCPULimit.Enabled)
```
Comment (ptodev) [inline component/otelcol/receiver/vcenter/vcenter_test.go:188]:
```suggestion
	require.False(t, otelArgs.Metrics.VcenterClusterCPUEffective.Enabled)
	require.True(t, otelArgs.Metrics.VcenterClusterCPULimit.Enabled)
```

I think this is clearer, but I'm happy if you leave it as it is.

Comment (marctc) [reply component/otelcol/receiver/vcenter/vcenter_test.go:188]: done

File: docs/sources/flow/reference/components/otelcol.receiver.vcenter.md [hunk 2]
```diff
@@ -78,11 +78,15 @@ The following blocks are supported inside the definition of
 Hierarchy | Block | Description | Required
 --------- | ----- | ----------- | --------
 tls | [tls][] | Configures TLS for the HTTP client. | no
+metrics | [metrics][] | Configures the metrics that this component generates to monitor its state. | no
+resource_attributes | [resource_attributes][] | Configures the resource attributes that this component generates to annotate telemetry data. | no
```
Comment (ptodev) [inline docs/sources/flow/reference/components/otelcol.receiver.vcenter.md:82]:
```suggestion
metrics | [metrics][] | Configures which metrics will be sent to downstream components. | no
resource_attributes | [resource_attributes][] | Configures resource attributes for metrics sent to downstream components. | no
```
`metrics` is not for the component's state :) It's the state of the vcenter endpoint.

File: docs/sources/flow/reference/components/otelcol.receiver.vcenter.md [hunk 3]
```diff
@@ -92,6 +96,77 @@ isn't provided, TLS won't be used for connections to the server.
 
 {{< docs/shared lookup="flow/reference/components/otelcol-tls-config-block.md" source="agent" version="<AGENT VERSION>" >}}
 
+### metrics block
+
+Name | Type | Description | Default | Required
```
Comment (ptodev) [inline docs/sources/flow/reference/components/otelcol.receiver.vcenter.md:101]: Do you mind making `metric` in the `Type` column a hyperlink to `#### metric block` please?
BTW After we get consensus on #5764, I might need to rework this doc to be in line with that future desicion.

**Review by ptodev — APPROVED** [review]
LGTM. You could address [the hyperlink commen](https://github.com/grafana/agent/pull/5813/files#r1399475809)t and the other comments from #5715 in a different PR. I think the docs now look pretty good for a first release 👌

**Pull request merged**